targets/TARGET_NORDIC/TARGET_NRF5/nordic_critical.c

```diff
@@ -15,43 +15,17 @@
  * limitations under the License.
  */
 
-#include <stdint.h>                  // uint32_t, UINT32_MAX
-#include <assert.h>                  // uint32_t, UINT32_MAX
-#include "cmsis.h"
-#include "nrf_soc.h"
-#include "nrf_sdm.h"
-#include "nrf_nvic.h"
+#include <stdint.h>
+#include "app_util_platform.h"
 
-static uint8_t  _sd_state = 0;
-static volatile uint32_t _entry_count = 0;
+static uint8_t nordic_cr_nested = 0;
 
 void core_util_critical_section_enter()
 {
-    // if a critical section has already been entered, just update the counter
-    if (_entry_count) {
```

Comment:
The one of possibles scenerio for 'previous code 'is:
For routine A _enter_count  check could had been positive, then execution of routine can been interrupted by routine B which  would call core_util_critical_section_exit. In result after routine A executed core_util_critical_section_enter the critical section is not enabled - and the _entry_count value is equal to 1.

So _entry_count check-modify must be a atomic operation in order to mitigate this and similar scenarios.
This Is exactly how app_util_critical_region_ente call work.

@pan- , @0xc0170 
This is answer to:
> @nvlsianpu Could you explain what was the root cause ?